src/core/config/descriptor_providers/xlio_config_schema.json

```diff
@@ -0,0 +1,1454 @@
+                                },
+                                "tx_completion_batch_size": {
+                                    "type": "integer",
+                                    "default": 64,
+                                    "minimum": 1,
+                                    "maximum": 64,
+                                    "title": "TX WRE completion batch size",
+                                    "description": "Number of TX WREs used until a completion signal is requested. Maps to XLIO_TX_WRE_BATCHING environment variable. Tuning this parameter allows a better control of the jitter encountered from the Tx CQE handling. Setting a high batching value results in high PPS and lower average latency. Setting a low batching value results in lower latency std-dev.\nValue range is 1-64"
+                                },
+                                "tx_max_inline_size": {
+                                    "type": "integer",
+                                    "default": 204,
+                                    "minimum": 0,
+                                    "maximum": 884,
+                                    "title": "Max TX inline size",
+                                    "description": "Maximum data size sent inline. Setting to 0 disables inlining. Maps to XLIO_TX_MAX_INLINE environment variable. Max send inline data set for QP. Data copied into the INLINE space is at least 32 bytes of headers and the rest can be user datagram payload. XLIO_TX_MAX_INLINE=0 disables INLINEing on the Tx transmit path. In older releases this parameter was called: XLIO_MAX_INLINE."
+                                },
+                                "rx_global_array_size": {
+                                    "type": "integer",
+                                    "default": 16000,
+                                    "minimum": 0,
+                                    "title": "RX WRE global array size",
+                                    "description": "Number of Work Request Elements allocated in all RQs. Maps to XLIO_RX_WRE environment variable. "
+                                },
+                                "rx_batch_size": {
+                                    "type": "integer",
+                                    "default": 1024,
+                                    "minimum": 1,
+                                    "maximum": 1024,
+                                    "title": "RX WRE batch size",
+                                    "description": "Number of Work Request Elements and RX buffers to batch before recycling. Maps to XLIO_RX_WRE_BATCHING environment variable. Batching decrease latency mean, but might increase latency STD.\nValue range is 1-1024."
+                                }
+                            }
+                        },
+                        "tcp_segments": {
```

Comment:
this one is not work element. maybe should go under buffers at least. or even protocol.tcp

Reply:
it's under buffers indeed and not work_request_elements - I understand the confusion though in such a large schema!